src/component/button/PermalinkButton/PermalinkButton.tsx

```diff
@@ -0,0 +1,167 @@
+import * as React from 'react';
+
+import SimpleButton from '@terrestris/react-geo/dist/Button/SimpleButton/SimpleButton';
+import Window from '@terrestris/react-geo/dist/Window/Window';
+
+import { TooltipPlacement } from 'antd/lib/tooltip';
+import Button, { ButtonProps } from 'antd/lib/button';
+import { message } from 'antd';
+
+import { PermalinkUtil } from '../../../util/PermalinkUtil/PermalinkUtil';
+
+import './PermalinkButton.css';
+
+
+interface DefaultPermalinkButtonProps {
+  type: 'default' | 'primary' | 'ghost' | 'dashed' | 'danger' | 'link';
+  shape: 'circle' | 'round';
+  iconName: string;
+}
+
+interface BaseProps {
+  map: any;
+  tooltip: string;
+  tooltipPlacement: TooltipPlacement;
+  t: (arg: string) => string;
+  getLink?: () => string;
+}
+
+interface PermalinkButtonState {
+  winVisible: boolean;
+}
+
+export type PermalinkButtonProps = BaseProps & Partial<DefaultPermalinkButtonProps> & ButtonProps;
+
+/**
+ * Class representing the PermalinkButton.
+ *
+ * @class PermalinkButton
+ * @extends React.Component
+ */
+export default class PermalinkButton extends React.Component<PermalinkButtonProps, PermalinkButtonState> {
```

Comment:
Even if other components in the base client are still `Components` I suggest to gradually move to function components by writing new components as such.